src/NHibernate/Proxy/DynamicProxy/ProxyCacheEntry.cs

```diff
@@ -24,6 +24,7 @@ public ProxyCacheEntry(System.Type baseType, System.Type[] interfaces)
 			}
 			BaseType = baseType;
 			Interfaces = interfaces ?? new System.Type[0];
+			_uniqueInterfaces = new HashSet<System.Type>(Interfaces);
```

Comment:
[`SetEquals`](https://msdn.microsoft.com/en-us/library/dd412096(v=vs.110).aspx) is not sensible to order.

> This method ignores the order of elements and any duplicate elements in other.

So I do not expect any order mismatch to cause an issue.